edk2toollib/acpi/ivrs_parser.py

```diff
@@ -0,0 +1,643 @@
+                                       header_byte_array[IVRS_TABLE.DeviceTableEntryLengthShort:])
+                (Type,
+                 self.EndDeviceID,
+                 _) = struct.unpack_from(IVRS_TABLE.DEVICE_TABLE_ENTRY.struct_format, header_byte_array[self.Length:])
+                if Type != 4:
+                    print("Start of range does not follow end of range")
+                    sys.exit(-1)
+                self.Length += IVRS_TABLE.DeviceTableEntryLengthShort
+            elif self.Type == 70:
+                self.TypeString = "Extended Select"
+                self.Length = IVRS_TABLE.DeviceTableEntryLengthLong
+                (self.ExtendedDTESetting,) = \
+                    struct.unpack_from("=I", header_byte_array[IVRS_TABLE.DeviceTableEntryLengthShort:])
+            elif self.Type == 71:
+                self.TypeString = "Extended Range"
+                self.Length = IVRS_TABLE.DeviceTableEntryLengthLong
+                (self.ExtendedDTESetting,) =\
+                    struct.unpack_from("=I", header_byte_array[IVRS_TABLE.DeviceTableEntryLengthShort:])
+                (Type,
+                 self.EndDeviceID,
+                 _) = struct.unpack_from(IVRS_TABLE.DEVICE_TABLE_ENTRY.struct_format, header_byte_array[self.Length:])
+                if Type != 4:
+                    print("Start of range does not follow end of range")
+                    sys.exit(-1)
+                self.Length += IVRS_TABLE.DeviceTableEntryLengthShort
+            elif self.Type == 72:
+                self.TypeString = "Special Device"
+                self.Length = IVRS_TABLE.DeviceTableEntryLengthLong
+                (self.Handle, self.SourceDeviceID, self.Variety) =\
+                    struct.unpack_from(IVRS_TABLE.DEVICE_TABLE_ENTRY.struct_format,
+                                       header_byte_array[IVRS_TABLE.DeviceTableEntryLengthShort:])
+            elif self.Type == 240:
+                self.TypeString = "Variable Length ACPI HID Device"
+                (self.HID, self.CID, self.UIDFormat, self.UIDLength) =\
+                    struct.unpack_from("=8s8sBB", header_byte_array[IVRS_TABLE.DeviceTableEntryLengthShort:])
+                self.Length = IVRS_TABLE.DeviceTableEntryLengthVar + self.UIDLength
+                if self.UIDFormat == 0:
+                    self.UID = None
+                if self.UIDFormat == 1:
+                    self.UID = struct.unpack_from("=Q", header_byte_array[IVRS_TABLE.DeviceTableEntryLengthVar:])
+                elif self.UIDFormat == 2:
+                    self.UID =\
+                        struct.unpack("=%ss" % self.UIDLength,
+                                      header_byte_array[IVRS_TABLE.DeviceTableEntryLengthVar:])
+            else:
+                print("Unknown Reserved Device Scope Type Found %d" % self.Type)
+                sys.exit(-1)
+
+            self.data = header_byte_array[:self.Length]
+
+        def set_xml(self, xml_item):
```

Comment:
This is the only object that has this pattern. Is there a reason/use-case for it?

Reply:
This is changed to align with other EncodeXmlNode methods.